src/main/java/ml/compute/InputData.java

```diff
@@ -0,0 +1,28 @@
+package ml.compute;
+
+import java.util.Set;
+
+import ml.preprocessor.types.PreprocessedData;
+import ml.preprocessor.types.RawData;
+
+/**
+ * Created by mikeecb on 17/10/2015.
+ */
+public class InputData {
```

Comment:
You are right, that should absolutely be an option. But I think that should just be another sub class of the inputdata. And then we might want to use the visitor pattern on input data?